intel-sgx/enclave-runner/src/tcs.rs

```diff
@@ -183,29 +190,40 @@ pub(crate) fn coenter<T: Tcs>(
                 panic!("Error entering enclave (VDSO): ret = {:?}, run = {:?}", std::io::Error::from_raw_os_error(-ret), run);
             }
         } else {
-            asm!("
-                    lea 1f(%rip), %rcx // set SGX AEP
-                    xchg {0}, %rbx
-1:                  enclu
-                    xchg %rbx, {0}
-                ",
-                inout(reg) tcs.address() => _, // rbx is used internally by LLVM and cannot be used as an operand for inline asm (#84658)
-                inout("eax") Enclu::EEnter as u32 => sgx_result,
-                out("rcx") _,
-                inout("rdx") p3,
-                inout("rdi") p1,
-                inout("rsi") p2,
-                inout("r8") p4,
-                inout("r9") p5,
-                inout("r10") debug_buf_ptr => _,
-                out("r11") _,
-                options(nostack, att_syntax)
-            );
+            while !exiting.load(Ordering::SeqCst) && enclu_leaf != (Enclu::EExit as u32) {
+                asm!("
+                        lea 1f(%rip), %rcx // set SGX AEP
+                        push %rbx          // store original rbx value on the stack
+                        mov {0}, %rbx
+                        enclu
+1:                      pop %rbx           // restore original rbx value in case we interrupt enclave during AEX
```

Comment:
I don't understand the change here. This looks like it's intended for RIP rewriting, but the current signal handler is only enabled with the VDSO.

Reply:
first of all,  AEP is changed here to check `exiting` flag on each AEX and interrupt enclu if needed.
secondly, push/pop instructions for RBX just allow for preserving the original value in RBX.

Reply:
rdx comments removed